nodeutil/node_test.go

```diff
@@ -71,6 +71,36 @@ func TestReflectBasics(t *testing.T) {
 	fc.AssertEqual(t, "bye", app.C.L)
 }
 
+func TestReflectOnReadNilValue(t *testing.T) {
+	mstr := `module x {
+		container c {
+			leaf l {
+				type string;
+			}
+		}
+	}`
+	m, err := parser.LoadModuleFromString(nil, mstr)
+	fc.RequireEqual(t, nil, err)
+
+	app := &reflectTestApp{
+		C: nil,
+	}
+	n := &Node{
+		Object: app,
+		OnRead: func(ref *Node, m meta.Definition, t reflect.Type, v reflect.Value) (reflect.Value, error) {
+			return v, nil
+		},
+	}
+	b := node.NewBrowser(m, n)
+
+	defer func() {
+		if r := recover(); r != nil {
+			t.Errorf("The code panics: %v", r)
+		}
+	}()
+	WriteJSON(b.Root())
```

Comment:
Primary test is exercising code, but can an assertion that can be made here on the output of WriteJSON or the accompanying err?

Reply:
Done.